ca_tools/scripts/robot_controller.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python
+
+import rospy
+from sensor_msgs.msg import Image
+from geometry_msgs.msg import Twist
+from std_msgs.msg import Bool
+
+
+class RobotController:
+
+    _FORWARD_VEL = 0.28
+    _ANGULAR_VEL = 0.32
+
+    def __init__(self):
+
+        self.twist_publisher = rospy.Publisher('/create1/cmd_vel', Twist, queue_size=10)
+
+        self.left_sensor_subscriber = rospy.Subscriber(
+            "/color_sensor_plugin/left_sensor", Bool, self.callback_left)
+        self.right_sensor_subscriber = rospy.Subscriber(
+            "/color_sensor_plugin/right_sensor", Bool, self.callback_right)
+
+        self.left_cam_subscriber = rospy.Subscriber(
+            "/left_cam/image_raw", Image, self.dummy_callback_left)
+
+        self.right_cam_subscriber = rospy.Subscriber(
+            "/right_cam/image_raw", Image, self.dummy_callback_right)
+
+        self.left_sensor_data = False
+        self.right_sensor_data = False
+
+        self.stuck_timer = rospy.Time.now()
+        self.stuck_flag = False
+
+    def callback_left(self,data):
+        self.left_sensor_data = data.data
+
+    def callback_right(self,data):
+        self.right_sensor_data = data.data
+
+    def dummy_callback_left(self,data):
+        pass
+
+    def dummy_callback_right(self,data):
+        pass
+
+    def publish_velocity(self, _linear, _angular):
```

Comment:
you could set default values for `_linear` and `_angular`, so you could call `self.publish_velocity()` and it would stop the robot, also, you could call `self.publish_velocity( _linear=self.LINEAR_VEL)`, and you wouldn't need to pass 2 parameters when you only use one.